pysat/tests/test_utils_io.py

```diff
@@ -278,23 +271,35 @@ def test_read_netcdf4_w_epoch_kwargs(self, kwargs, target):
 
         """
 
-        # TODO(#947) Expand to xarray objects once functionality is updated
-        if not self.testInst.pandas_format:
-            pytest.skip('Not yet implemented for xarray')
-
         # Create a bunch of files by year and doy
-        outfile = os.path.join(self.testInst.files.data_path,
-                               'pysat_test_ncdf.nc')
+        outfile = os.path.join(self.tempdir.name,
+                               'pysat_{:}_ncdf.nc'.format(self.testInst.name))
         self.testInst.load(date=self.stime, use_header=True)
+        if not self.testInst.pandas_format:
+            # A distinct epoch name must be used for the saved xarray dataset.
+            test_epoch = 'test_epoch'
+            kwargs['epoch_name'] = test_epoch
+            # Xarray testing requires conversion to integer with ms default
+            self.testInst.data[test_epoch] = \
+                self.testInst[self.epoch_name].astype('int64') / 1.e6
+        else:
+            if 'epoch_name' in kwargs:
+                kwargs.pop('epoch_name')
+
         io.inst_to_netcdf(self.testInst, fname=outfile)
 
         # Load the data that was created
         kwargs['pandas_format'] = self.testInst.pandas_format
         self.loaded_inst, meta = io.load_netcdf(outfile, **kwargs)
 
         # Check that the step size is expected
-        default_delta = (self.testInst.index[1] - self.testInst.index[0])
-        loaded_delta = (self.loaded_inst.index[1] - self.loaded_inst.index[0])
+        if self.testInst.pandas_format:
+            default_delta = np.diff(self.testInst.index[:2])
+            loaded_delta = np.diff(self.loaded_inst.index[:2])
+        else:
+            default_delta = np.diff(self.testInst[self.epoch_name][:2])
+            # Average over 4 deltas to prevent rounding errors
```

Comment:
```suggestion
            default_delta = np.diff(self.testInst[self.epoch_name][:2])

            # Average over 4 deltas to prevent rounding errors
```